nep-29.mediawiki

```diff
@@ -0,0 +1,60 @@
+<pre>
+  NEP: 29
+  Title: Contract deployment/update callback function
+  Author: Roman Khimov <roman@nspcc.ru>
+  Type: Standard
+  Status: Accepted
+  Created: 2024-07-18
+</pre>
+
+==Abstract==
+
+This proposal standardizes the callback function that can be implemented by contracts to have some code executed right after initial deployment or update.
+
+==Motivation==
+
+Contracts may need to run some initialization/update code right after their deployment or upgrade. Neo N3 has provided a mechanism for that since version 3.0, but it was never properly standardized, so this proposal fixes it.
+
+==Specification==
+
+Neo N3 has a native <code>ContractManagement</code> contract that is used for contract deployment and update via <code>deploy</code> and <code>update</code> methods of it. After settling with the new NEF and manifest both of them can invoke a special <code>_deploy</code> method if it's implemented by the contract.
```

Comment:
We always said only N3, I think that we can remove Neo from the name

Reply:
Well, I've seen "Neo N3" used a lot and maybe it's even more important now with "Neo X".